sklearn/tests/test_discriminant_analysis.py

```diff
@@ -417,27 +478,56 @@ def test_qda():
     # Classes should have at least 2 elements
     assert_raises(ValueError, clf.fit, X6, y4)
 
+    # Number of sample should be bigger than number of classes
+    assert_raises(ValueError, clf.fit, X8, y8)
```

Comment:
please use `with pytest.raises(ValueError, match=...):` construct instead. We're removing all `assert_raises` instances now.

Reply:
I made this change but not through all the code. Only the parts that I have introduced. Same for array_almost_equal.